tools/topology/topology1/sof-tgl-max98357a-rt5682.m4

```diff
@@ -146,7 +146,12 @@ define(DMIC_PIPELINE_KWD_ID, `12')
 define(DMIC_DAI_LINK_16k_ID, eval(BOARD_DMIC_BE_ID_BASE + 1))
 define(DMIC_PIPELINE_48k_CORE_ID, `1')
 
-ifdef(`RTNR', `define(`DMICPROC', rtnr)', `')
+# 5ms period is required for RTNR build 20220728 and later versions
+ifdef(`RTNR',
```

Comment:
Just a reminder that these definitions would be overwritten by GOOGLE_RTC_AUDIO_PROCESSING if defined, which is not mutually exclusive to RTNR. Other than that, this looks good to me.

Reply:
ok, so RTNR and GOOGLE_RTC are not mutually exclusive. @johnylin76 I guess they are both using DMICPROC ? If so are they using different DMICs or the same DMIC at runtime. Can you give a bit more context. Thanks

Reply:
they are both on DMIC48K pipeline if applied:
PCM <--- RTNR <--- GOOGLE_RTC <--- DMIC48K

so actually here is a missing case: both RTNR and GOOGLE_RTC_AUDIO_PROCESSING are defined. We should rephrase line 149~159 as following:
```
ifdef(`GOOGLE_RTC_AUDIO_PROCESSING',
	`ifdef(`RTNR',
		`define(`DMICPROC', google-rtc-audio-processing-rtnr)',
		`define(`DMICPROC', google-rtc-audio-processing)')'
	`define(`DMIC_48k_PERIOD_US', 10000)'
	,
	`ifdef(`RTNR',
		`define(`DMICPROC', rtnr)'
		`define(`DMIC_48k_PERIOD_US', 5000)')'
)
```

Reply:
@MingJenTai can you update as suggested by @johnylin76 . Thanks.

Reply:
@johnylin76 Thanks for the suggestions. Changes are updated.

Reply:
Hello @johnylin76 Is this PR ok for you?

Reply:
LGTM thanks